File: 125-valid-palindrome/valid-palindrome.cs
public class Solution {
    public bool IsPalindrome(string s) {
        

        int first = 0;
        int last = s.Length-1;

        while(first<=last)
        {
             char  start = s[first];
             char end = s[last];
             if(!char.IsLetterOrDigit(start))
             {
                first++;
             }else if(!char.IsLetterOrDigit(end))
             {
                last--;

             }
             else{
                if(char.ToLower(start)!=(char.ToLower(end))){
                    return false;

                }
                first++;
                last--;

             }

        }
        return true;








        
    }
}

File: 125-valid-palindrome/valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {

        int first = 0;
        int last = s.length() - 1;
        while (first <= last) {
            char start = s.charAt(first);
            char end = s.charAt(last);
            if (!Character.isLetterOrDigit(start)) {
                first++;

            } else if (!Character.isLetterOrDigit(end)) {
                last--;
            } else {
                if (Character.toLowerCase(start) != Character.toLowerCase(end)) {
                    return false;
                }

                first++;
                last--;
            }
        }
        return true;

    }
}